src/lib/paginate.js

```diff
@@ -0,0 +1,70 @@
+import mkdirp from 'mkdirp'
+import log from './logs'
+import saveFiles from './saveFiles'
+import fetchData from './fetchData'
+import cloneDeep from 'lodash.clonedeep'
+
+export default function paginate (bundleName, fetchOptions, _config) {
+  const config = {
+    savePath: './data',
+    ..._config
+  }
+
+  /**
+   * Create directory (synchronously)
+   */
+  mkdirp(config.savePath, function (err) {
+    if (err) return console.error(err)
+    log.success(`${config.savePath} created successfully!`)
+  })
+
+  /**
+   * Fetch all languages
+   */
+   if (config.languages && config.languages.length) {
+     Promise.all(
+       config.languages.map(language => fetchData(config, language, fetchOptions))
+     ).then((datas) => {
+       // for each lang save files
+       datas.forEach(saveDataToFile)
+       log.success('DONE.')
+     })
+   } else {
+     fetchData(config, false, fetchOptions).then((data) => {
+       // if one language, only save this lang
+       saveDataToFile(data)
+       log.success('DONE.')
+     })
+   }
+
+   /**
+    * Save files (in this case paginated)
+    */
+  function saveDataToFile (data) {
+    const dataClone = cloneDeep(data) // deep clone the data in order to do calculations
+    const itemCount = _config.itemsPerPage ? _config.itemsPerPage : 10 // how many items per page, default 10
+    const itemsTotal = dataClone[bundleName].length // itemsTotal - items length
+    const slice = Math.ceil(itemsTotal / itemCount) // round up slices (101 items will be 11 pages – last page with 1 item)
+    let from = 0
+    for (var i = 1; i <= slice; i++) {
+      saveFiles(
```

Comment:
We need to "collect" all these calls into an array and resolve it. (see annotation in bundle.js)